NetCore/NetCoreRepro/ProxyFactory.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Castle.DynamicProxy;
+using System;
+
+namespace NetCoreRepro
+{
+    public static class ProxyFactory
+    {
+        static ProxyGenerator generator = new ProxyGenerator();
+        static Interceptor interceptor = new Interceptor();
+
+        public static T CreateProxy<T>() where T : class
+        {
+            return generator.CreateInterfaceProxyWithoutTarget<T>(interceptor);
+        }
+
+        private sealed class Interceptor : IInterceptor
```

Comment:
This DynamicProxy `IInterceptor` essentially replaces Moq's `mock.Setup(m => m.Method(...)).Returns(true)`.